src/Build/BackEnd/Components/RequestBuilder/IntrinsicTasks/ItemGroupIntrinsicTask.cs

```diff
@@ -100,6 +104,20 @@ internal override void ExecuteTask(Lookup lookup)
                                 }
                             }
 
+                            if (!String.IsNullOrEmpty(child.MatchOnMetadata))
+                            {
+                                var matchOnMetadataEvaluated = bucket.Expander.ExpandIntoStringListLeaveEscaped(child.MatchOnMetadata, ExpanderOptions.ExpandAll, child.MatchOnMetadataLocation).ToList();
+                                if (matchOnMetadataEvaluated.Count > 0)
+                                {
+                                    matchOnMetadata = new HashSet<string>(matchOnMetadataEvaluated);
+                                }
+                            }
+
+                            if (Enum.TryParse(child.MatchOnMetadataOptions, out MatchOnMetadataOptions options))
```

Comment:
Since this is on a very hot path, could we move this under the previous if? I imagine there aren't any cases when String.IsNullOrEmpty(child.MatchOnMetadata) but Enum.TryParse(child.MatchOnMetadataOptions, out MatchOnMetadataOptions options).